### PR TITLE
[ESIMD] Add clamp API

### PR DESCRIPTION
**Pull request by fineg74**

### CODE REVIEW

File: sycl/include/sycl/ext/intel/esimd/math.hpp [hunk 1]
```diff
@@ -1851,6 +1851,39 @@ __ESIMD_API uint64_t rdtsc() {
...
+/// Performs clamping of values in a vector between min and max values.
+/// @tparam T type of the vectors.
+/// @tparam N size of the vectors.
+/// @param src vector containing an input.
+/// @param min vector containing minimum values.
+/// @param max vector containing maximum values.
+/// @return vector containing clamped input values between minimum and maximum.
+template <typename T, int N>
+__ESIMD_API __ESIMD_NS::simd<T, N> clamp(__ESIMD_NS::simd<T, N> src,
+                                         __ESIMD_NS::simd<T, N> min,
```
Comment (v-klochkov) [inline sycl/include/sycl/ext/intel/esimd/math.hpp:1863]: I would not name the parameters `min` and `max` after having those issues with min and max macros on Windows defined in some system headers and disable-able by NOMINMAX.

File: sycl/include/sycl/ext/intel/esimd/math.hpp [hunk 2]
```diff
@@ -1851,6 +1851,39 @@ __ESIMD_API uint64_t rdtsc() {
...
+                                         __ESIMD_NS::simd<T, N> max) {
+  __ESIMD_NS::simd<T, N> Result = src;
+  Result.merge(min, src < min);
+  Result.merge(max, src > max);
+  return Result;
+}
+
+/// Performs clamping of values in a vector between min and max values.
+/// The variant of the API having scalars for minimum and maximum values.
```
Comment (sarnex) [inline sycl/include/sycl/ext/intel/esimd/math.hpp:1872]:
```suggestion
/// This variant of the API uses scalars for minimum and maximum values.
```

**Review by sarnex — APPROVED** [review]
lgtm minor comment

**Review by v-klochkov — APPROVED** [review]
Looks good, but just one more minor comment: If it is user-facing API, then it should use C++ style using underscores, e.g.: `min_val`

**Pull request merged**